### DISCUSSION TITLE
devtools::install error

### DISCUSSION THREAD

**Discussion by Rawizard** [IBAHCM/RPiR#333] [Archive] [answered]
Hi @richardreeve, @soniamitchell,

I keep getting this error when I try to use devtools::install and I have absolutely no idea what to do about it. Any help would be much appreciated!!

ERROR: failed to lock directory ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library’ for modifying
Try removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/00LOCK-RawizardDiversity’
Error in (function (command = NULL, args = character(), error_on_status = TRUE,  : 
  System command 'R' failed, exit status: 3, stdout & stderr were printed
Type .Last.error.trace to see where the error occured

Stack trace:

 1. devtools::install()
 2. pkgbuild::with_build_tools(required = FALSE, callr::rcmd("INSTALL",  ...
 3. callr::rcmd("INSTALL", c(install_path, opts), echo = !quiet,  ...
 4. callr:::run_r(options)
 5. base:::with(options, with_envvar(env, do.call(processx::run,  ...
 6. base:::with.default(options, with_envvar(env, do.call(processx::run,  ...
 7. base:::eval(substitute(expr), data, enclos = parent.frame())
 8. base:::eval(substitute(expr), data, enclos = parent.frame())
 9. callr:::with_envvar(env, do.call(processx::run, c(list(bin, args = real_cmdargs,  ...
 10. base:::force(code)
 11. base:::do.call(processx::run, c(list(bin, args = real_cmdargs,  ...
 12. (function (command = NULL, args = character(), error_on_status = TRUE,  ...
 13. throw(new_process_error(res, call = sys.call(), echo = echo,  ...

 x System command 'R' failed, exit status: 3, stdout & stderr were printed 

Cheers,

Jamie

**Comment by soniamitchell** [comment, marked as answer]
Hi @Rawizard, this is the key bit:

> Try removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/00LOCK-RawizardDiversity’

Try removing that file, restarting R, and running `devtools:install()` again 👍

**Reply by Rawizard** [reply]
Ok thanks, is there a way to search for files? As I cant find that file in any of the obvious places.

**Reply by soniamitchell** [reply]
Are you using a mac or windows?

**Reply by Rawizard** [reply]
Mac, thanks.

**Reply by soniamitchell** [reply]
Open any folder, right click on the title of the folder and go to Macintosh HD, Library is in there 👍

**Reply by Rawizard** [reply]
That worked! Absolute legend thank you, meltdown avoided haha

**Reply by soniamitchell** [reply]
Awesome! 🦊
